### PR TITLE
sys/tsrb: add peek functions

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adds peek functions to `tsrb`.

### Testing procedure

Unit tests provided.

### CODE REVIEW

**Review by benpicco — APPROVED** [review]
looks good to me

File: tests/unittests/tests-tsrb/tests-tsrb.c [hunk 1]
```diff
@@ -115,6 +115,43 @@ static void test_get(void)
...
+        TEST_ASSERT_EQUAL_INT(0, tsrb_add_one(&_tsrb, TEST_INPUT + i));
+    }
+    TEST_ASSERT_EQUAL_INT(BUFFER_SIZE, tsrb_peek(&_tsrb, _io_buffer,
+                                                sizeof(_io_buffer)));
+    for (int i = 0; i < BUFFER_SIZE; i++) {
+        TEST_ASSERT_EQUAL_INT((TEST_INPUT + i), _io_buffer[i]);
+    }
+    for (int i = BUFFER_SIZE; i < (int)sizeof(_io_buffer); i++) {
+        TEST_ASSERT_EQUAL_INT(IO_BUFFER_CANARY, _io_buffer[i]);
+    }
```
Comment (aabadie) [inline tests/unittests/tests-tsrb/tests-tsrb.c:151]: Maybe use `unsigned` ?

```suggestion
    for (unsigned i = 0; i < BUFFER_SIZE; i++) {
        TEST_ASSERT_EQUAL_INT(0, tsrb_add_one(&_tsrb, TEST_INPUT + i));
    }
    TEST_ASSERT_EQUAL_INT(BUFFER_SIZE, tsrb_peek(&_tsrb, _io_buffer,
                                                sizeof(_io_buffer)));
    for (unsigned i = 0; i < BUFFER_SIZE; i++) {
        TEST_ASSERT_EQUAL_INT((TEST_INPUT + i), _io_buffer[i]);
    }
    for (unsigned i = BUFFER_SIZE; i < sizeof(_io_buffer); i++) {
        TEST_ASSERT_EQUAL_INT(IO_BUFFER_CANARY, _io_buffer[i]);
    }
```

Comment (fjmolinas) [reply tests/unittests/tests-tsrb/tests-tsrb.c:151]: I mostly copied over the other tests, so if this was to be changed I would do it for the others as well...

**Review by kaspar030 — APPROVED** [review]
LGTM to me as well.

**Pull request merged**